plugins/destination/mongodb/client/client.go

```diff
@@ -0,0 +1,51 @@
+package client
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cloudquery/plugin-sdk/plugins/destination"
+	"github.com/cloudquery/plugin-sdk/specs"
+	"github.com/rs/zerolog"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type Client struct {
+	destination.UnimplementedUnmanagedWriter
+	destination.DefaultReverseTransformer
+	logger     zerolog.Logger
+	spec       specs.Destination
+	pluginSpec Spec
+	client     *mongo.Client
+}
+
+func New(ctx context.Context, logger zerolog.Logger, destSpec specs.Destination) (destination.Client, error) {
+	var err error
+	c := &Client{
+		logger: logger.With().Str("module", "mongo-dest").Logger(),
+		spec:   destSpec,
+	}
+
+	var spec Spec
+	if err := destSpec.UnmarshalSpec(&spec); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal MongoDB spec: %w", err)
+	}
+	if err := spec.Validate(); err != nil {
+		return nil, err
+	}
+	c.client, err = mongo.NewClient(options.Client().ApplyURI(spec.ConnectionString))
+	if err != nil {
+		return nil, err
+	}
+	if err := c.client.Connect(context.Background()); err != nil {
```

Comment:
Should this be `context.Background()`, or `ctx`?

Reply:
I think the same thing as with `BigQuery` it stores the context and runs checks in the background.